d2/src/main/java/com/linkedin/d2/discovery/stores/zk/ZooKeeperEphemeralStore.java

```diff
@@ -476,6 +476,12 @@ private void setRawClientTrackingNode()
     String rawD2ClientTrackingPath = D2Utils.getRawClientTrackingPath();
     try
     {
+      if (rawD2ClientTrackingPath.contains("post-commit"))
```

Comment:
also would be good to add a doc about why this check is needed with a sample path like "export-content-data-multiproduct-post-commit-mpdep-i001-workspace-credential-store_7c08d85e507e41cab5598191ca6d459ccbce3f89-credential-store-credential-store-integ-test"